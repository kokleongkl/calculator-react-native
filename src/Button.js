import React from "react";
import { View, TouchableHighlight, Text } from "react-native";
import { styles } from "./styles";

const Button = React.memo((props) => (
  <TouchableHighlight
    style={[styles.buttonStyles, props.style]}
    onPress={props.onPress}
    underlayColor="#777"
  >
    <View style={styles.containerButton}>
      <Text style={styles.textButtons}>{props.text}</Text>
    </View>
  </TouchableHighlight>
));

export default Button;
