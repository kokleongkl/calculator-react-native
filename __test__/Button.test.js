import React from "react";
import Button from "../src/Button";
import renderer from "react-test-renderer";

describe("Button", () => {
  describe("Button Component rendered", () => {
    it("should match to snapshot", () => {
      const testcomponent = renderer.create(<Button text={"="} />).toJSON();
      expect(testcomponent).toMatchSnapshot();
    });
  });
});
