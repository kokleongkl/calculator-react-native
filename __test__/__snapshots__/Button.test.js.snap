// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "flex": 1,
        "margin": 4,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 26,
        }
      }
    >
      =
    </Text>
  </View>
</View>
`;
